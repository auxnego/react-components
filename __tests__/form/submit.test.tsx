import React from "react";
import {create} from "react-test-renderer";

import Submit from "../../form/submit";

it("normal submit", () => {

  expect(create(
    <Submit>Enviar</Submit>
  ).toJSON()).toMatchSnapshot();

});


it("loading submit", () => {

  expect(create(
    <Submit loading={true}>Cargando</Submit>
  ).toJSON()).toMatchSnapshot();

});


it("normal submit", () => {

  expect(create(
    <Submit 
      setClass={{loading:"cargando-class"}} 
      loading={true}
      loadingLabel="Esperando Datos"
    >Enviar</Submit>
  ).toJSON()).toMatchSnapshot();

});

