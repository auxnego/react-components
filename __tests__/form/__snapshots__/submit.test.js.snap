// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading submit 1`] = `
<button
  className="button button--disabled button__submit--loading"
  disabled={true}
  onClick={[Function]}
  role="button"
>
  <i
    className="fa fa-spinner animate-spin"
  />
  Cargando
</button>
`;

exports[`normal submit 1`] = `
<button
  className="button button__submit"
  disabled={false}
  onClick={[Function]}
  role="button"
>
  Enviar
</button>
`;

exports[`normal submit 2`] = `
<button
  className="button button--disabled button__submit--loading"
  disabled={true}
  onClick={[Function]}
  role="button"
>
  <i
    className="fa fa-spinner animate-spin"
  />
  Esperando Datos
</button>
`;
